inc/classes/class-srm-redirect.php

```diff
@@ -98,6 +98,28 @@ public function match_redirect( $requested_path ) {
 			$normalized_requested_path = $requested_path;
 		}
 
+		// normalise the request path with and without query strings, for comparison later
+		$requested_query_params             = '';
+		$normalized_requested_path_no_query = false;
+
+		if ( function_exists( 'wp_parse_url' ) ) {
+			$parsed_requested_path = wp_parse_url( $normalized_requested_path );
+		} else {
+			$parsed_requested_path = parse_url( $normalized_requested_path );
+		}
+
+		if ( is_array( $parsed_requested_path ) ) {
+			$normalized_requested_path_no_query = '';
+			$requested_query_params             = '';
+
+			if ( ! empty( $parsed_requested_path['path'] ) ) {
+				$normalized_requested_path_no_query = untrailingslashit( stripslashes( $parsed_requested_path['path'] ) );
```

Comment:
`$parsed_requested_path['path']` will be always exist because the `#normalized_requested_path` variable is not empty.

Reply:
Updated.

Reply:
This is unfortunately not correct, as our debug log is getting flooded with undefined index warnings: `This is unfortunately not correct, as our debug log is getting flooded with undefined index warnings. We're not yet sure how it comes to `

We're not yet sure how what's wrong exactly, but it's clear that there's an issue here.